### PR TITLE
chore: add eslint plugin for react hooks

### LINKED ISSUE #379: eslint support for react hooks
### Description
With React 16.8, React Hooks are available in a stable release. An [eslint plugin](https://www.npmjs.com/package/eslint-plugin-react-hooks) is available to validate the use of hooks within a component.

---

_**NOTE:** Where applicable, please include uncropped screen captures._

_**DISCLAIMER:**
After triaging an issue, the fundamental-react team will see if it can be reproduced or confirmed. If more information is needed, the fundamental-react team will contact the author.  Any issues awaiting responses from the author for more than 7 days will be closed.  The author can re-open the issue at a later time if they can present the requested information._

### PR DESCRIPTION
**Pull request by chrismanciero**
### Description
Adds eslint checking for use of react hooks within components

Reference sites:

- [https://reactjs.org/blog/2019/02/06/react-v16.8.0.html](https://reactjs.org/blog/2019/02/06/react-v16.8.0.html)
- [https://reactjs.org/docs/hooks-rules.html](https://reactjs.org/docs/hooks-rules.html)
- [https://www.npmjs.com/package/eslint-plugin-react-hooks](https://www.npmjs.com/package/eslint-plugin-react-hooks)

fixes #379

### CODE REVIEW

File: .eslintrc [hunk 1]
```diff
@@ -17,6 +17,7 @@ plugins:
     - jsx-a11y
     - loosely-restrict-imports
     - react
+    - react-hooks
```
Comment (greg-a-smith) [inline .eslintrc]: Not that this is a bad idea, but are we using React Hooks?

**Comment by bcullman** [comment]
i'd like this in our back-pocket, but I suggest closing until hooks lands in fundamental-react. that will be when a) code is written, and b) this

https://github.com/SAP/fundamental-react/blob/master/package.json#L127-L128

is bumped to `^16.8.0`

File: .eslintrc (hunk 1 above)
Comment (chrismanciero) [reply .eslintrc]: I added this more as a future check. Hooks was just released with react v16.8. A new developer can check out our library and make/change a component using hooks.
Obviously we have to approve it :)

**Comment by chrismanciero** [comment]
Doesn't `"react": "^16.6.3",` in our package.json file mean that if someone was to do an npm install of our library it will install react version > 16.6.3 (potentially 16.8) or does the npm install limit it to just 16.6.* versions?

**Comment by greg-a-smith** [comment]
@chrismanciero The hat (`^`) means the consumer is willing to take any additional minor or patch-level version bumps, but not major versions (breaking changes).  That said, the `package-lock.json` file may prevent this since its job is to "lock" the version so the environment is repeatable.  You can remove that file, run `npm install` and have it be regenerated at which time, it would bump to the newest non-breaking version.  Hope that helps.

**Comment by chrismanciero** [comment]
Yes it does thanks. closing the pr.

**Pull request closed without merging**